pyproject.toml

```
@@ -22,7 +22,20 @@ classifiers=[
     "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
-    # insert python dependencies as list here
+  "numpy",
+  "dem_stitcher>=2.4.0",
+  "rasterio",
+  "shapely",
+  "jinja2",
+  "asf_search>=6.4.0",
+  "geopandas",
+  "gdal",
+  "s3fs",
+  "pyarrow",
+  "hyp3lib>=2,<3",
+  # FIXME specifying either of the below dependencies causes the build to fail
```

Turns out we didn't have any of the dependencies in the pyproject.toml. I've added all of the packages. Unfortunately when either `isce2` or `opencv` is uncommented, installing hyp3-isce2 with the command:

`python -m pip install .`

results in the error:
```
Processing /Users/ffwilliams2/Repositories/hyp3/hyp3-isce2
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (1.23.5)
Requirement already satisfied: dem-stitcher>=2.4.0 in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (2.5.0)
Requirement already satisfied: rasterio in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (1.3.6)
Requirement already satisfied: shapely in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (2.0.1)
Requirement already satisfied: jinja2 in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (3.1.2)
Requirement already satisfied: asf-search>=6.4.0 in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (6.7.1)
Requirement already satisfied: geopandas in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (0.14.1)
Requirement already satisfied: gdal in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (3.6.3)
Requirement already satisfied: s3fs in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (2023.10.0)
Requirement already satisfied: pyarrow in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (12.0.1)
Requirement already satisfied: hyp3lib<3,>=2 in /Users/ffwilliams2/mambaforge/envs/hyp3_isce2_2/lib/python3.11/site-packages (from hyp3-isce2==0.8.2.dev20+g3db074b.d20231117) (2.0.2)
INFO: pip is looking at multiple versions of hyp3-isce2 to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement opencv (from hyp3-isce2) (from versions: none)
ERROR: No matching distribution found for opencv
```
My guess is that this is an incredibly annoying packaging issue and my preference is to let this sleeping dog lie for now.

---

@forrestfwilliams I'm pretty sure that opencv and isce2 are not distributed on pypi, and thus can't be included here:
* https://pypi.org/project/opencv/
* https://pypi.org/project/isce2/
```suggestion
  # Conda-forge only dependencies are listed below
```

---

ohhhh